### DISCUSSION TITLE
App deployed to root domain

### DISCUSSION THREAD

**Discussion by gleatd01** [caprover/caprover#1398] [Q&A]
Hello all,
I may just simply be missing the instruction. If CapRover has been installed at captain.domain.com how do I deploy an app to domain.com?

**Comment by githubsaturn** [comment]
1- set up an app, like `app1`
2- your app will be available as `app1.domain.com`
3- go to your dashboard and add `domain.com` 

![image](https://user-images.githubusercontent.com/10582082/165196024-4d4b50a4-fbb5-4796-a515-95f05da37db4.png)

**Reply by graphiostudio** [reply]
When you "add domain" it works but it looks like it only forwards that custom domain to the app on the root domain. So, after you enter domain, URL changes to the root app. Is it possible that app root domain is not visible on the url but instead its completely the custom domain added on the "add domain"?

**Reply by githubsaturn** [reply]
It's not CapRover that does that... Are you deploying a wordpress application or something? Those applications have an internal redirect mechanism.... If you're using wordpess, just go to settings and edit the URL
![image](https://user-images.githubusercontent.com/10582082/177796213-d4e3c22f-0fa5-4cfe-9abe-87c6aa030770.png)

**Reply by graphiostudio** [reply]
Omg, you're right. I was playing around and it worked with node apps, worked with everything else I throw at. I came back to report and then saw your post. I didn't know WP does that. Thanks a lot! Used caprover only 1 day and god damn, what a time saver! Incredible! Huge hugs to the developers :)

**Reply by orhan6808** [reply]
> 1- set up an app, like `app1` 2- your app will be available as `app1.domain.com` 3- go to your dashboard and add `domain.com`
> 
> ![image](https://user-images.githubusercontent.com/10582082/165196024-4d4b50a4-fbb5-4796-a515-95f05da37db4.png)

hello how did you removed captain from the root domain?

**Reply by githubsaturn** [reply]
your root domain is `something.com` or `another.something.com` - there is no `captain` in the root domain. 

`captain` is only present in the dashboard domain which is `captain.<root-domain>`

**Comment by hydego17** [comment]
Hi, I have similar issue, but now its working since I added my root domain to the app configuration.

`App: client.mydomain.com`
`Custom Domain: mydomain.com`

Everything works fine, except for `www` redirection.

**Problem:** 
---
When I try to curl my website, for example:
`curl -I https://mydomain.com`, or `curl -I www.mydomain.com`
Everything works just fine.


But when i try to curl with specific format like `https://www`:
`curl -I https://www.mydomain.com`,
I get an error saying something like:
`curl: (60) SSL: no alternative certificate subject name matches target host name 'www.mydomain.com' `,

I also notice that my site with this specific link is not crawlable.

I've tried to tweak the nginx configuration with something as follow, but still not working:
```
server {
    listen       80;
    server_name  www.mydomain.com;
    return 301 https://mydomain.com$request_uri;
}

server {
    listen       80;
    server_name  www.client.mydomain.com;
    return 301 https://client.mydomain.com$request_uri;
}

<%
if (s.forceSsl) {
%>
    server {
...    
```


Is there any workaround to solve this @githubsaturn ? Thank you for ur amazing work btw

**Reply by githubsaturn** [reply]
You'll just have to enable the HTTPs for your `www` domain
